src/fireedge/src/modules/features/General/slice.js

```diff
@@ -0,0 +1,324 @@
+
+            const arrayAtPath = get(state.modifiedFields, pathWithoutIndex, [])
+            const index = parseInt(lastSegment, 10)
+            while (arrayAtPath.length <= index) {
+              arrayAtPath.push({})
+            }
+            if (payload?.__flag__ === 'DELETE') {
+              // Filter the array to mark the correct position with delete
+              arrayAtPath.filter((item) => !item.__delete__)[
+                index
+              ].__delete__ = true
+            } else {
+              // In array cases, we need to calculate the correction index (could exists elements that were deleted)
+              const arrayWithoutDeleteItems = arrayAtPath.filter(
+                (item) => !item.__delete__
+              )
+              const arrayWithoutDeleteItemsLength =
+                arrayWithoutDeleteItems.length
+
+              // Update original item
+              if (index < arrayWithoutDeleteItemsLength) {
+                arrayWithoutDeleteItems[index] = merge(
+                  arrayWithoutDeleteItems[index] || {},
+                  mergedPayload
+                )
+              } else {
+                // Create new item
+                arrayAtPath[arrayAtPath.length] = mergedPayload
+              }
+            }
+
+            set(state.modifiedFields, pathWithoutIndex, arrayAtPath)
+          } else {
+            const nested = pathSegments?.length > 2
+            const nestedBasePath = nested
+              ? pathSegments.slice(0, -1).join('.')
+              : pathSegments.join('.')
+            const nestedKey = pathSegments.slice(-1)[0]
+
+            let existingNestedValue = get(
+              state.modifiedFields,
+              nestedBasePath,
+              {}
+            )
+
+            if (nested) {
+              existingNestedValue[nestedKey] = merge(
+                {},
+                existingNestedValue || {},
+                get(mergedPayload, pathSegments[0], {})
+              )
+            } else {
+              existingNestedValue = merge(
+                {},
+                existingNestedValue || {},
+                get(mergedPayload, pathSegments[0], {})
+              )
+            }
+
+            const parsedExisting = parsePayload(existingNestedValue, fieldPath)
+            set(state.modifiedFields, nestedBasePath, parsedExisting)
+          }
+        } else {
+          state.modifiedFields = merge({}, state.modifiedFields, mergedPayload)
+        }
+
+        if (payload?.setPath) {
+          state.fieldPath = payload.setPath
+        }
+      })
+      .addCase(actions.resetModifiedFields, (state) => {
+        state.modifiedFields = {}
+      })
+
+      /* UPLOAD NOTIFICATION */
+      .addCase(actions.setUploadSnackbar, (state, { payload }) => ({
+        ...state,
+        upload: payload,
+      }))
+      /* NOTIFICATION ACTIONS */
+      .addCase(actions.enqueueSnackbar, (state, { payload }) => {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).